src/NSrtm.Core/Pgm/DataDesciption/PgmDataDescriptionExtractor.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace NSrtm.Core.Pgm
+{
+    internal static class PgmPreamble
+    {
+        public static readonly IReadOnlyDictionary<string, Regex> Extractor =
+            new Dictionary<string, Regex>
+            {
+                {"MagicNumber", new Regex(@"^(P5)[^$]", RegexOptions.Multiline)},
```

Comment:
This magic number should be checked as `preamble.Substring(2) = "P5"`, not by looking for all line starts (I think it pwuld match file where second line is "P5".